spec/models/user_spec.rb

```diff
@@ -190,18 +339,138 @@
         expect(subject).to receive(:is_manager).and_return(false)
         expect(subject).to receive(:is_viewer).and_return(false)
         expect(subject).to receive(:permitted_apos).and_return([1])
-        expect(subject.can_view_something?).to be_truthy
+        expect(subject.can_view_something?).to be true
+      end
+    end
+  end
+
+  describe '#can_manage_object?' do
+    let(:druid) { 'druid:hv992ry2431' }
+    let(:item) { instantiate_fixture(druid, Dor::AdminPolicyObject) }
+    before :each do
+      expect(subject).not_to receive(:is_viewer)
+    end
+    it 'returns false' do
+      expect(subject.can_manage_object?(item)).to be false
+    end
+    context 'when admin' do
+      it 'returns true' do
+        expect(subject).to receive(:is_admin).and_return(true)
+        expect(subject).not_to receive(:is_manager)
+        expect(subject.can_manage_object?(item)).to be true
+      end
+    end
+    context 'when manager' do
+      it 'returns true' do
+        expect(subject).to receive(:is_admin).and_return(false)
+        expect(subject).to receive(:is_manager).and_return(true)
+        expect(subject.can_manage_object?(item)).to be true
+      end
+    end
+    context 'when viewer' do
+      it 'returns false' do
+        expect(subject).to receive(:is_admin).and_return(false)
+        expect(subject).to receive(:is_manager).and_return(false)
+        allow(item).to receive(:can_manage_item?).and_return(false)
+        allow(item).to receive(:can_manage_content?).and_return(false)
+        expect(subject.can_manage_object?(item)).to be false
+      end
+    end
+    context 'when user has an authorized role' do
+      it 'returns true' do
+        allow(subject).to receive(:is_admin).and_return(false)
+        allow(subject).to receive(:is_manager).and_return(false)
+        expect(item).not_to receive(:can_manage_content?)
+        manager_roles = ["dor-administrator", "sdr-administrator"]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.